conmon-rs/server/src/rpc.rs

```diff
@@ -105,19 +115,30 @@ impl conmon::Server for Server {
 
         Promise::from_future(async move {
             match child_reaper
-                .create_child(&runtime, &args, console, pidfile)
+                .create_child(&runtime, &args, Some(&console), pidfile)
                 .await
             {
                 Ok(grandchild_pid) => {
                     let mut resp = results.get().init_response();
                     // register grandchild with server
                     let child = Child::new(id, grandchild_pid, child.exit_paths);
-                    let _ = child_reaper.watch_grandchild(child);
+                    capnp_err!(child_reaper.watch_grandchild(child))?;
+
                     // TODO return the grandchild exit code
-                    // TODO return the stdout and stderr of the child
+
+                    let mut stdout = String::new();
+                    for msg in console.message_rx() {
+                        match msg {
+                            Message::Data(s) => stdout.push_str(&s),
+                            Message::Done => break,
+                        }
+                    }
+
+                    resp.set_stdout(&stdout);
                     resp.set_exit_code(0);
```

Comment:
@rphillips @haircommander have you discussed stderr handling when using the runc file descriptor passing? I thought we could not distinguish stdout from stderr in that case.

Reply:
Ah, I wasn't aware of that... We should remove the stderr components.

Reply:
Good, I'll prepare that.